aten/src/ATen/native/cuda/MultiMarginLoss.cu

```diff
@@ -248,21 +250,25 @@ Tensor& multi_margin_loss_cuda_out(
               margin);
           C10_CUDA_KERNEL_LAUNCH_CHECK();
         }
-        at::sum_out(out, tmp_output, /*dims=*/IntArrayRef{});
+        out = at::sum(tmp_output, /*dims=*/IntArrayRef{});
+        need_resize_ = true;
       }
     }
   });
 
   if (!out.is_alias_of(out_)) {
     out_.copy_(out);
```

Comment:
this doesn't seem to be the correct fix. `out_` at this point is the `nframe` size, and `copy_` fills all the unneeded values, and only `resize_` then saves the day. We should avoid this.

Reply:
Agreed, was there a historical version why the CPU logic for resizing didn't work? Seems to work for the user-submitted case and in `test_scalar_check`.

Reply:
I don't know, sorry, I didn't look at the cpu implementation